### PR TITLE
Fix pure ipaddreses (ip's without a port) could not resolve.

### LINKED ISSUE #134: Auto add 25565 as port if none is set for servers in serverlist

### PR DESCRIPTION
**Pull request by kepet19**
Fixes  #134
normal hostnames could be resolve without a port. it was only Ip adresses that could not. 

The name you can see in the server list is the same as the adress.
### Now
image

I have simplified the code and it can still connect to server where it is protected by [TCPShield](https://tcpshield.com/).
I did throw in a empty server where the field is empty. 
also kkmp.dk does not exits. 

FIY this pr also removes  regex

### CODE REVIEW

**Comment by terrarier2111** [comment]
Why would you want to remove the srv resolution? it is needed to join the vast majority of big server (e g. hypixel)

**Comment by kepet19** [comment]
> why would you want to remove the srv resolution?

Upps I thought it was a normal dns lookup where it got the ip's. but I now see it can also get ports. 
So the solution I made would not function if they are not using default minecraft port. 
I now see it can be a problem. 

FIY. mc.hypixel.net seems to function properly.

**Comment by terrarier2111** [comment]
Yea, so just revert your changes and only use the default port if the ip is numerical and no port was provided (or if the srv lookup fails) OR as an alternative you could first try the result of the srv lookup, then the provided port (if there is one) and at the or the default port, the second approach would be more complete although a bit harder to implement

**Comment by kepet19** [comment]
I made a new commit, that keeps srv
not sure if we should remove the regex crate.
rust std has a ip parser. 
```rust
   if let Err(_) = std::net::IpAddr::from_str(&addres) {
     ...
   }
   
   same almost
   
  if !IPADDRESS_PATTERN.is_match(&addres) {
    ...
  }
  
  ```

**Comment by terrarier2111** [comment]
> I made a new commit, that keeps srv
> not sure if we should remove the regex crate.
> rust std has a ip parser.
> 
> ```rust
>    if let Err(_) = std::net::IpAddr::from_str(&addres) {
>      ...
>    }
>    
>    same almost
>    
>   if !IPADDRESS_PATTERN.is_match(&addres) {
>     ...
>   }
>   
> ```

If you are able to drop it without dropping any functionality that would be great. Are you sure rust can check if somthing is a domain or an ip tho?

**Comment by kepet19** [comment]
yes I am sure 

it is define like this. 
 
```rust
enum IpAddr {
ipV4
ipV6
}

**Comment by terrarier2111** [comment]
So you are also sure it that it doesn't automatically resolve the ip behind the domain?

**Comment by kepet19** [comment]
> So you are also sure it that it doesn't automatically resolve the ip behind the domain?
Here is the for IpAddr
https://doc.rust-lang.org/std/net/enum.IpAddr.html

Yes I tripped checked the code and also run some Manuel test with the  server browser.

File: protocol/src/protocol/mod.rs [hunk 1]
```diff
@@ -1116,59 +1114,63 @@ pub struct Conn {
...
+                    return Ok(conn);
+                }
+            }
+        }
+
+        debug!("{}'s ip may be {}:{}.", target, address, port);
+        Conn::try_stream(&address, port, protocol_version)
+    }
+
+    fn try_stream(addres: &str, port: u16, protocol_version: i32) -> Result<Conn, Error> {
```
Comment (terrarier2111) [inline protocol/src/protocol/mod.rs:1168]: there is a typo, in "addres" it should be "address"

Comment (kepet19) [reply protocol/src/protocol/mod.rs:1168]: good catch!

File: protocol/src/protocol/mod.rs [hunk 2]
```diff
@@ -1116,59 +1114,63 @@ pub struct Conn {
...
-        if !IPADDRESS_PATTERN.is_match(target) {
-            debug!("{} has an no address! :(", address);
-            let result = Conn::get_server_addresses(target);
-            // TODO: Try all possible ips not just the first!
-            let next = result.iter().next().unwrap();
-            debug!("{}'s ip may be {}.", address, next);
-            address = next.to_string();
+        let mut port = 25565;
+
+        if target.contains(':') {
```
Comment (terrarier2111) [inline protocol/src/protocol/mod.rs:1141]: You could split_once directly here and use a if let instead of first checking contains and then splitting

Comment (kepet19) [reply protocol/src/protocol/mod.rs]: I don't even know why, I did not think about that xD.

**Comment by terrarier2111** [comment]
Can you rebase? then i'll merge this

**Comment by terrarier2111** [comment]
Okay, can you rebase once again? then i'll merge it lol

**Comment by kepet19** [comment]
This pr it is rebased on the lastest commit  on master  ca6e8e2 ?

**Comment by terrarier2111** [comment]
There are merge conflicts @kepet19

**Comment by kepet19** [comment]
Fixed!

**Comment by terrarier2111** [comment]
> Fixed! @terrarier2111

I'll wait for the ci and as i have no access to a computer in like 10 minutes until i get home i'll merge it once i got home.

**Comment by terrarier2111** [comment]
Okay, i am home now, can you resolve the merge conflicts? its the last time now, i promise

**Comment by kepet19** [comment]
it is running again 
no problem!

**Pull request merged**